src/components/Api.js

```diff
@@ -0,0 +1,145 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._authorization = headers.authorization;
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  updateProfileInfo({ title, description }) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: title, // API expects title or name?
+        about: description,
+      }),
+    })
+      .then((res) => {
+        return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

This code is repeated in every request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the `reference` for the method rather than the call. 

**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Hi, for the 'could be improved'  portion- thank you for the suggestion. I wanted to see if I could achieve this. I did an example and commented out lines 18-30. Can you confirm if I did this correctly?

Reply:
> Can you confirm if I did this correctly?

Yes, it's correct

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

  updateProfileInfo({ title, description }) {
    return this._request(`${this._baseUrl}/users/me`, {
      method: "PATCH",
      headers: this._headers,
      body: JSON.stringify({
        name: title,
        about: description,
      }),
    });
  }

```

Reply:
So, you can replace all `fetch` the same way